src/components/ToastNotificationList.tsx

```diff
@@ -0,0 +1,239 @@
+import { Icon, Notification, ValueOf } from "@canonical/react-components";
+import {
+  DefaultTitles,
+  NotificationSeverity,
+} from "@canonical/react-components/dist/components/Notification/Notification";
+import {
+  GroupedNotificationCount,
+  ToastNotificationType,
+} from "context/toastNotificationProvider";
+import React, { FC, useLayoutEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+import { iconLookup, severityOrder } from "util/notifications";
+import Animate from "./Animate";
+import { usePrefersReducedMotion } from "util/usePreferReducedMotion";
+
+export type FilterTypes = ValueOf<typeof NotificationSeverity>;
+
+interface Props {
+  notifications: ToastNotificationType[];
+  onDismiss: (notification?: ToastNotificationType[]) => void;
+  groupedCount: GroupedNotificationCount;
+  show: boolean;
+}
+
+const ToastNotificationList: FC<Props> = ({
+  notifications,
+  onDismiss,
+  groupedCount = {},
+  show,
+}) => {
+  const [filters, setFilters] = useState<Set<FilterTypes>>(new Set());
+  const prevNotificationsSize = useRef<number>(notifications.length);
+  const containerRef = useRef<HTMLUListElement>(null);
+  const hasFilters = !!filters.size;
+  const preferReducedMotion = usePrefersReducedMotion();
+
+  useLayoutEffect(() => {
```

Comment:
@lorumic or @aaryanporwal , when you have some time could you review the content of this `useLayoutEffect`? The purpose for this here is to adjust the scroll height of the list container if a new item gets added when the scroll bar is at its initial top position. This makes the scroll visually stable, something requested by piper.